File: skills/desktop-app-dev/clients/media_client.go
package media

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io"
	"net/http"
)

type MediaClient struct {
	BaseURL string
	HTTP    *http.Client
	Token   string
}

func NewMediaClient(baseURL string, token ...string) *MediaClient {
	client := &MediaClient{BaseURL: baseURL, HTTP: http.DefaultClient}
	if len(token) > 0 {
		client.Token = token[0]
	}
	return client
}

func (c *MediaClient) Enqueue(kind string, payload any, dedupeKey string, priority int) (map[string]any, error) {
	body, _ := json.Marshal(map[string]any{
		"kind": kind, "payload": payload, "dedupe_key": dedupeKey, "priority": priority,
	})
	req, err := http.NewRequest("POST", c.BaseURL+"/tasks", bytes.NewReader(body))
	if err != nil {
		return nil, err
	}
	req.Header.Set("Content-Type", "application/json")
	c.auth(req)
	resp, err := c.HTTP.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()
	var result map[string]any
	if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
		return nil, err
	}
	return result, nil
}

func (c *MediaClient) Task(id int64) (map[string]any, error) {
	req, err := http.NewRequest("GET", fmt.Sprintf("%s/tasks/%d", c.BaseURL, id), nil)
	if err != nil {
		return nil, err
	}
	c.auth(req)
	resp, err := c.HTTP.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()
	data, err := io.ReadAll(resp.Body)
	if err != nil {
		return nil, err
	}
	var result map[string]any
	if err := json.Unmarshal(data, &result); err != nil {
		return nil, err
	}
	return result, nil
}

func (c *MediaClient) DepsProgress() (map[string]any, error) {
	req, err := http.NewRequest("GET", c.BaseURL+"/deps/progress", nil)
	if err != nil {
		return nil, err
	}
	c.auth(req)
	resp, err := c.HTTP.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()
	data, err := io.ReadAll(resp.Body)
	if err != nil {
		return nil, err
	}
	var result map[string]any
	if err := json.Unmarshal(data, &result); err != nil {
		return nil, err
	}
	return result, nil
}

func (c *MediaClient) DepsStatus() (map[string]any, error) {
	req, err := http.NewRequest("GET", c.BaseURL+"/deps/status", nil)
	if err != nil {
		return nil, err
	}
	c.auth(req)
	resp, err := c.HTTP.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()
	data, err := io.ReadAll(resp.Body)
	if err != nil {
		return nil, err
	}
	var result map[string]any
	if err := json.Unmarshal(data, &result); err != nil {
		return nil, err
	}
	return result, nil
}

func (c *MediaClient) InstallDeps() (map[string]any, error) {
	req, err := http.NewRequest("POST", c.BaseURL+"/deps/install", nil)
	if err != nil {
		return nil, err
	}
	c.auth(req)
	resp, err := c.HTTP.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()
	data, err := io.ReadAll(resp.Body)
	if err != nil {
		return nil, err
	}
	var result map[string]any
	if err := json.Unmarshal(data, &result); err != nil {
		return nil, err
	}
	return result, nil
}

func (c *MediaClient) auth(req *http.Request) {
	if c.Token != "" {
		req.Header.Set("Authorization", "Bearer "+c.Token)
	}
}

File: skills/desktop-app-dev/clients/media_client.rs
use serde_json::{json, Value};

pub struct MediaClient {
    base_url: String,
    client: reqwest::Client,
    token: Option<String>,
}

impl MediaClient {
    pub fn new(base_url: impl Into<String>, token: Option<String>) -> Self {
        Self {
            base_url: base_url.into(),
            client: reqwest::Client::new(),
            token,
        }
    }

    pub async fn enqueue(
        &self,
        kind: &str,
        payload: Value,
        dedupe_key: Option<&str>,
    ) -> Result<Value, reqwest::Error> {
        let body = json!({"kind": kind, "payload": payload, "dedupe_key": dedupe_key});
        let mut request = self
            .client
            .post(format!("{}/tasks", self.base_url))
            .json(&body);
        if let Some(token) = &self.token {
            request = request.bearer_auth(token);
        }
        request
            .send()
            .await?
            .json()
            .await
    }

    pub async fn task(&self, id: i64) -> Result<Value, reqwest::Error> {
        let mut request = self.client.get(format!("{}/tasks/{}", self.base_url, id));
        if let Some(token) = &self.token {
            request = request.bearer_auth(token);
        }
        request
            .send()
            .await?
            .json()
            .await
    }

    pub async fn deps_progress(&self) -> Result<Value, reqwest::Error> {
        let mut request = self
            .client
            .get(format!("{}/deps/progress", self.base_url));
        if let Some(token) = &self.token {
            request = request.bearer_auth(token);
        }
        request
            .send()
            .await?
            .json()
            .await
    }

    pub async fn deps_status(&self) -> Result<Value, reqwest::Error> {
        let mut request = self
            .client
            .get(format!("{}/deps/status", self.base_url));
        if let Some(token) = &self.token {
            request = request.bearer_auth(token);
        }
        request
            .send()
            .await?
            .json()
            .await
    }

    pub async fn install_deps(&self) -> Result<Value, reqwest::Error> {
        let mut request = self
            .client
            .post(format!("{}/deps/install", self.base_url));
        if let Some(token) = &self.token {
            request = request.bearer_auth(token);
        }
        request
            .send()
            .await?
            .json()
            .await
    }
}
